packages/smart-wallet/src/proposals/upgrade-walletFactory-proposal.js

```diff
@@ -0,0 +1,252 @@
+  const walletReviver = await E(ppKit.creatorFacet).getWalletReviver();
+  const newPrivateArgs = harden({
+    storageNode,
+    walletBridgeManager,
+    walletReviver,
+  });
+
+  console.log(
+    'upgradeWalletFactory: upgrading with newPrivateArgs',
+    newPrivateArgs,
+  );
+  await E(wfKit.adminFacet).upgradeContract(
+    walletFactoryRef.bundleID,
+    newPrivateArgs,
+  );
+  console.log('upgradeWalletFactory: done');
+};
+harden(upgradeWalletFactory);
+
+/**
+ * @param { BootstrapPowers } powers
+ */
+export const publishAgoricBrandsDisplayInfo = async ({
+  consume: { agoricNames, board, chainStorage },
+}) => {
+  // @ts-expect-error chainStorage is only falsy in testing
+  const boardAux = E(chainStorage).makeChildNode(BOARD_AUX);
+  const publishBrandInfo = async brand => {
+    const [id, displayInfo, allegedName] = await Promise.all([
+      E(board).getId(brand),
+      E(brand).getDisplayInfo(),
+      E(brand).getAllegedName(),
+    ]);
+    const node = E(boardAux).makeChildNode(id);
+    const aux = marshalData.toCapData(harden({ allegedName, displayInfo }));
+    await E(node).setValue(JSON.stringify(aux));
+  };
+
+  /** @type {ERef<NameHub>} */
+  const brandHub = E(agoricNames).lookup('brand');
+  const brands = await E(brandHub).values();
+  // tolerate failure; in particular, for the timer brand
+  await Promise.allSettled(brands.map(publishBrandInfo));
+};
+harden(publishAgoricBrandsDisplayInfo);
+
+/** @type { import("@agoric/vats/src/core/lib-boot").BootstrapManifest } */
+const manifest = {
+  [upgradeWalletFactory.name]: {
+    // include rationale for closely-held, high authority capabilities
+    consume: {
+      contractKits: `to upgrade walletFactory using its adminFacet`,
+      governedContractKits:
+        'to get walletReviver from provisionPool.creatorFacet',
+      chainStorage: 'to allow walletFactory to (continue) write to vstorage',
+      walletBridgeManager: 'to handle bridged cosmos SpendAction messages',
+    },
+    // widely-shared, low authority instance handles need no rationale
+    instance: {
+      consume: { walletFactory: true, provisionPool: true },
+    },
+  },
+  [publishAgoricBrandsDisplayInfo.name]: {
+    consume: { agoricNames: true, board: true, chainStorage: true },
+  },
+};
+harden(manifest);
+
+export const getManifestForUpgrade = (_powers, { walletFactoryRef }) => {
+  return harden({
+    manifest,
+    options: { walletFactoryRef },
+  });
+};
+
+// avoid importing all of ERTP
+const makeAmount = (brand, value) => harden({ brand, value });
+
+const IST_UNIT = 1_000_000n;
+const CENT = IST_UNIT / 100n;
+
+/**
+ * Make a storage node for auxilliary data for a value on the board.
+ *
+ * @param {ERef<StorageNode>} chainStorage
+ * @param {string} boardId
+ */
+const makeBoardAuxNode = async (chainStorage, boardId) => {
+  const boardAux = E(chainStorage).makeChildNode(BOARD_AUX);
+  return E(boardAux).makeChildNode(boardId);
+};
```

Comment:
Is anything awaiting results from `makeBoardAuxNode` via `publishBrandInfo`? Even if not, it would probably make sense to call `E(chainStorage).makeChildNode(BOARD_AUX)` only once rather than once for every brand.

Reply:
I thought I fixed that. Weird.